src/CarbonAware.Aggregators/src/CarbonAware/CarbonAwareAggregator.cs

```diff
@@ -74,6 +75,22 @@ public async Task<IEnumerable<EmissionsForecast>> GetCurrentForecastDataAsync(ID
         }
     }
 
+    /// <inheritdoc />
+    public async Task<double> CalculateAverageCarbonIntensityAsync(IDictionary props)
+    {
+        using (var activity = Activity.StartActivity())
+        {
+            var start = GetOffsetOrDefault(props, CarbonAwareConstants.Start, DateTimeOffset.Now.ToUniversalTime());
```

Comment:
For calculating an average, start and end time are required.  I think our implementation should reflect that.

I made a similar comment on https://github.com/microsoft/carbon-aware-sdk/pull/110#discussion_r929909410 that we may want to add a `GetOffsetOrThrow` method for required params, or use another pattern like https://github.com/microsoft/carbon-aware-sdk/pull/110#discussion_r929907731 to manage aggregator input validation

If you decide to go this direction I would recommend syncing w/ @juzuluag @JenMadiedo and @pritipath to coordinate

Reply:
well, today the required params are enforced in the request contract, at the controller level -- but I think you're proposing "defense in depth" by having the aggregator also enforce their presence? 
I don't see a `GetOffsetOrThrow` in PR 110, so let me sync offline

Reply:
@dnastrain I think they went a different direction and created a props validator method.

But beyond simply "defense in depth", we have the expectation that the WebAPI controllers are not the only codepath into the aggregator (the CLI being the other today, but the design is not limited to those 2).   For that reason, we can't rely solely on controller checks to do the enforcement.